flashinfer/gdn_kernels/delta_rule_dsl/custom_compile_cache.py

```
@@ -50,16 +68,173 @@ def _compile_options_key(options):
+
+def _assemble_sm120a_cubin(ptx: str) -> bytes:
+    with tempfile.TemporaryDirectory(prefix="flashinfer_dsl_tma_patch_") as tmp_dir:
+        ptx_path = os.path.join(tmp_dir, "kernel.ptx")
+        cubin_path = os.path.join(tmp_dir, "kernel.cubin")
+        with open(ptx_path, "w", encoding="utf-8") as f:
+            f.write(ptx)
+
+        cuda_path = os.environ.get("CUDA_PATH", "/usr/local/cuda")
+        ptxas = os.path.join(cuda_path, "bin", "ptxas")
+        cmd = [ptxas, "-arch=sm_120a", ptx_path, "-o", cubin_path]
+        result = subprocess.run(cmd, check=False, text=True, capture_output=True)
```

Ruff (0.15.15)

[error] `subprocess` call: check for execution of untrusted input

(S603)